pkg/neotest/coverage.go

```diff
@@ -0,0 +1,153 @@
+package neotest
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/nspcc-dev/neo-go/pkg/compiler"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+)
+
+var rawCoverage = make(map[scriptHash]*scriptRawCoverage)
+
+var enabled = false
+var coverProfile = ""
+
+type scriptHash = util.Uint160
+
+type scriptRawCoverage struct {
+	debugInfo      *compiler.DebugInfo
+	offsetsVisited []int
+}
+
+type coverBlock struct {
+	startLine uint // Line number for block start.
+	startCol  uint // Column number for block start.
+	endLine   uint // Line number for block end.
+	endCol    uint // Column number for block end.
+	stmts     uint // Number of statements included in this block.
+	counts    uint
+}
+
+type documentName = string
+
+func isCoverageEnabled() bool {
+	if enabled {
+		return true
+	}
+	const coverProfileFlag = "test.coverprofile"
+	flag.VisitAll(func(f *flag.Flag) {
```

Comment:
The function name looks like it doesn't have any side-effects. What about doing this `flag.VisitAll` in `init()`. This way it will run once and set `enabled` variable.

Reply:
Flags are defined and parsed when `main` is run - after init blocks, so you can't use `init()` here.
Maybe we should just change name to something more clear.
![image](https://github.com/nspcc-dev/neo-go/assets/40753025/5ba3e8e0-92a4-40e3-ba87-0cd0508e8a0e)